src/app/shared/form/builder/parsers/complex-field-parser.ts

```diff
@@ -0,0 +1,125 @@
+import { Inject } from '@angular/core';
+import { FormFieldModel } from '../models/form-field.model';
+import { FormFieldMetadataValueObject } from '../models/form-field-metadata-value.model';
+import {
+  DsDynamicInputModel,
+  DsDynamicInputModelConfig
+} from '../ds-dynamic-form-ui/models/ds-dynamic-input.model';
+import {
+  DynamicFormControlLayout,
+} from '@ng-dynamic-forms/core';
+import {
+  COMPLEX_GROUP_SUFFIX,
+  COMPLEX_INPUT_SUFFIX,
+  DynamicComplexModel,
+  DynamicComplexModelConfig,
+
+} from '../ds-dynamic-form-ui/models/ds-dynamic-complex.model';
+import { hasValue, isNotEmpty } from '../../../empty.util';
+import { ParserOptions } from './parser-options';
+import {
+  CONFIG_DATA,
+  FieldParser,
+  INIT_FORM_VALUES,
+  PARSER_OPTIONS,
+  SUBMISSION_ID
+} from './field-parser';
+
+export class ComplexFieldParser extends FieldParser {
+
+  constructor(
+    @Inject(SUBMISSION_ID) submissionId: string,
+    @Inject(CONFIG_DATA) configData: FormFieldModel,
+    @Inject(INIT_FORM_VALUES) initFormValues,
+    @Inject(PARSER_OPTIONS) parserOptions: ParserOptions,
+    protected separator: string,
+    protected placeholders: string[]) {
+    super(submissionId, configData, initFormValues, parserOptions);
+    this.separator = separator;
+  }
+
+  public modelFactory(fieldValue?: FormFieldMetadataValueObject | any, label?: boolean): any {
+
+    let clsGroup: DynamicFormControlLayout;
+    let clsInput: DynamicFormControlLayout;
+    const id: string = this.configData.selectableMetadata[0].metadata;
+
+    clsGroup = {
+      element: {
+        control: 'form-row',
+      }
+    };
+
+    clsInput = {
+      grid: {
+        host: 'col-sm-12'
```

Comment:
?

Reply:
I have copied it from default DSpace code and have changed "host" value from 'col-sm-6' to 'col-sm-12'